emmet-core/emmet/core/qchem/calculation.py

```diff
@@ -0,0 +1,437 @@
+            esp=qcoutput.data["ESP"][-1],
+            resp=qcoutput.data["RESP"][-1],
+            nbo_data=qcoutput.data["nbo_data"],
+            frequencies=qcoutput.data["frequencies"],
+            frequency_modes=qcoutput.data.get(["frequency_mode_vectors"], []),
+            final_energy=qcoutput.data["final_energy"],
+            enthalpy=qcoutput.data["enthalpy"],
+            entropy=qcoutput.data["entropy"],
+            scan_energies=qcoutput.data.get(["scan_energies"]),
+            scan_geometries=qcoutput.data["optimized_geometries"]
+            if qcoutput.data["scan_energies"]
+            else None,
+            scan_molecules=qcoutput.data["molecules_from_optimized_geometries"]
+            if qcoutput.data["scan_energies"]
+            else None,
+            pcm_gradients=qcoutput.data["pcm_gradients"][0],
+            cds_gradients=qcoutput.data["CDS_gradients"][0],
+            dipoles=qcoutput.data["dipoles"],
+            gap_info=qcoutput.data["gap_info"],
+        )
+
+    # TODO What can be done for the trajectories, also how will walltime and cputime be reconciled
+
+
+class Calculation(BaseModel):
+    """Full QChem calculation inputs and outputs."""
+
+    dir_name: str = Field(None, description="The directory for this QChem calculation")
+    qchem_version: str = Field(
+        None, description="QChem version used to perform the current calculation"
+    )
+    has_qchem_completed: Union[QChemStatus, bool] = Field(
+        None, description="Whether QChem calculated the calculation successfully"
+    )
+    input: CalculationInput = Field(
+        None, description="QChem input settings for the calculation"
+    )
+    output: CalculationOutput = Field(
+        None, description="The QChem calculation output document"
+    )
+    completed_at: str = Field(
+        None, description="Timestamp for when the calculation was completed"
+    )
+    task_name: str = Field(
+        None,
+        description="Name of task given by custodian (e.g. opt1, opt2, freq1, freq2)",
+    )
+    output_file_paths: Dict[str, str] = Field(
+        None,
+        description="Paths (relative to dir_name) of the QChem output files associated with this calculation",
+    )
+    level_of_theory: LevelOfTheory = Field(
+        None,
+        description="Levels of theory used for the QChem calculation: For instance, B97-D/6-31g*",
+    )
+    task_type: TaskType = Field(
+        None,
+        description="Calculation task type like Single Point, Geometry Optimization. Frequency...",
+    )
+    calc_type: CalcType = Field(
+        None,
+        description="Combination dict of LOT + TaskType: B97-D/6-31g*/VACUUM Geometry Optimization",
+    )
+
+    @classmethod
+    def from_qchem_files(
+        cls,
+        dir_name: Union[Path, str],
+        task_name: str,
+        qcinput_file: Union[Path, str],
+        qcoutput_file: Union[Path, str],
+        store_energy_trajectory: bool = False,
```

Comment:
You should probably be able to tell if you need a trajectory based on whether or not it's an optimization job. Seems to me that opt/tsopt jobs should just about always have optimization trajectories associated.